addons/pos_mrp/models/pos_order.py

```diff
@@ -8,12 +8,18 @@ class PosOrderLine(models.Model):
 
     def _get_stock_moves_to_consider(self, stock_moves, product):
         self.ensure_one()
-        bom = product.env['mrp.bom']._bom_find(product, company_id=stock_moves.company_id.id, bom_type='phantom')[product]
+        bom = product.env['mrp.bom']._bom_find(product, company_id=stock_moves.company_id.id, bom_type='phantom').get(product)
         if not bom:
             return super()._get_stock_moves_to_consider(stock_moves, product)
         _dummy, components = bom.explode(product, self.qty)
         ml_product_to_consider = (product.bom_ids and [comp[0].product_id.id for comp in components]) or [product.id]
-        return stock_moves.filtered(lambda ml: ml.product_id.id in ml_product_to_consider and ml.bom_line_id)
+
+        for bom_line in bom.bom_line_ids:
+            nested_bom = product.env['mrp.bom']._bom_find(bom_line.product_id, company_id=stock_moves.company_id.id, bom_type='phantom').get(bom_line.product_id)
```

Comment:
It looks very similar to the previous code. It's already the purpose of explode to get all the nested boms (returned in _dummy)

Reply:
Thank you for pointing this out; I hadn’t noticed it before.  
I have updated my code, and the changes should now be reflected in the commit.